GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -405,11 +404,20 @@ public Process GetGitProcess(string command, string workingDirectory, string dot
 
             // Removing trace variables that might change git output and break parsing
             // List of environment variables: https://git-scm.com/book/gr/v2/Git-Internals-Environment-Variables
-            foreach (string key in processInfo.EnvironmentVariables.Keys.Cast<string>()
-                .Where(x => x.StartsWith("GIT_TRACE", StringComparison.OrdinalIgnoreCase))
-                .ToList())
+            foreach (string key in processInfo.EnvironmentVariables.Keys.Cast<string>())
             {
-                processInfo.EnvironmentVariables.Remove(key);
+                // If GIT_TRACE is set to a fully-rooted path, then Git sends the trace
+                // output to that path instead of stdout (GIT_TRACE=1) or stderr (GIT_TRACE=2).
+                if (key.StartsWith("GIT_TRACE", StringComparison.OrdinalIgnoreCase))
+                {
+                    string value = processInfo.EnvironmentVariables[key];
+
+                    if (value.IndexOfAny(Path.GetInvalidPathChars()) >= 0

```

Comment:
I am guessing most of the time the environment variable will not have invalid path characters.  Would it be better from a perf standpoint to just call `IsPathRooted` and catch `ArgumentException` (in case there is an invalid char)?

Reply:
It is almost always better to check a condition than to catch an exception.

Reply:
Not when the condition has a perf cost and will almost never yield a hit, whereas the exception handler is free to set up. I would favor the exception handler here as well.